core/src/parser/grammar.lalrpop

```diff
@@ -557,6 +557,30 @@ Match: Match = {
 // A default annotation in a pattern.
 DefaultAnnot: RichTerm = "?" <t: Term> => t;
 
+// A metadata keyword returned as an indent. In some positions, those are
```

Comment:
Can we do this for all (or most) of our keywords? I think even a `let` or `rec` should be unambiguous in field name positions.

Reply:
I would vote for doing this for metadata keyword only at first. Although I don't have right now any example of an existing keyword or a future extension that could cause an issue, but let's just give it a bit more thinking (metadata keyword are much more restricted in where they can appear). At worst we can do that in a follow-up PR.